Core/TextureReplacer.cpp

```diff
@@ -535,58 +565,56 @@ class TextureSaveTask : public Task {
 	}
 };
 
+bool TextureReplacer::WillSave(const ReplacedTextureDecodeInfo &replacedInfo) {
+	_assert_msg_(enabled_, "Replacement not enabled");
+	if (!g_Config.bSaveNewTextures)
+		return false;
+	// Don't save the PPGe texture.
+	if (replacedInfo.addr > 0x05000000 && replacedInfo.addr < PSP_GetKernelMemoryEnd())
+		return false;
+	if (replacedInfo.isVideo && !allowVideo_)
+		return false;
+
+	return true;
+}
 
 void TextureReplacer::NotifyTextureDecoded(const ReplacedTextureDecodeInfo &replacedInfo, const void *data, int pitch, int level, int w, int h) {
 	_assert_msg_(enabled_, "Replacement not enabled");
-	if (!g_Config.bSaveNewTextures) {
+	if (!WillSave(replacedInfo)) {
 		// Ignore.
 		return;
 	}
-	if (replacedInfo.addr > 0x05000000 && replacedInfo.addr < PSP_GetKernelMemoryEnd()) {
-		// Don't save the PPGe texture.
-		return;
-	}
-	if (replacedInfo.isVideo && !allowVideo_) {
+	if (ignoreMipmap_ && level > 0) {
 		return;
 	}
+
 	u64 cachekey = replacedInfo.cachekey;
 	if (ignoreAddress_) {
 		cachekey = cachekey & 0xFFFFFFFFULL;
 	}
-	if (ignoreMipmap_ && level > 0) {
-		return;
-	}
```

Comment:
This check is probably what saved us before, removing it resulted in #15492 .

I know what's wrong though, code to detect the fake mipmap situation isn't quite right, fixing.

Reply:
Ah right. Well I just had a quick skim of this one after actually finding the real root cause. No fault of this PR.